presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -399,11 +411,18 @@ public static Table fromMetastoreApiTable(org.apache.hadoop.hive.metastore.api.T
             throw new PrestoException(HIVE_INVALID_METADATA, "Table is missing storage descriptor");
         }
 
+        // Materialized view fetched from Metastore in Facebook uses MANAGED_TABLE table type. Cast it back to MATERIALIZED_VIEW here.
+        PrestoTableType tableType = PrestoTableType.optionalValueOf(table.getTableType()).orElse(OTHER);
+        if ("true".equals(table.getParameters().get(PRESTO_MATERIALIZED_VIEW_FLAG))) {
+            checkState(TableType.MANAGED_TABLE.name().equals(table.getTableType()), "Materialized view %s has incorrect table type %s from Metastore.", table.getTableName(), table.getTableType());
+            tableType = MATERIALIZED_VIEW;
+        }
+
         Table.Builder tableBuilder = Table.builder()
                 .setDatabaseName(table.getDbName())
                 .setTableName(table.getTableName())
                 .setOwner(nullToEmpty(table.getOwner()))
-                .setTableType(PrestoTableType.optionalValueOf(table.getTableType()).orElse(OTHER))
+                .setTableType(tableType)
```

Comment:
Accidental change? We don't have to modify this.

Reply:
Here's where we cast managed_table back to materialized_view for getTable().